misc/server/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM rust:1.66-bullseye as builder
+WORKDIR /usr/src/rust-libp2p-server
+
+# Run with access to the target cache to speed up builds
+WORKDIR /workspace
+ADD . .
+RUN --mount=type=cache,target=./target \
+    --mount=type=cache,target=/usr/local/cargo/registry \
+    cargo build --release --package libp2p-server
+
+RUN --mount=type=cache,target=./target \
+    mv ./target/release/libp2p-server /usr/local/bin/libp2p-server
+
+FROM debian:bullseye-slim
```

Comment:
Can it be something smaller like `distroless`?

Reply:
:+1: done